src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Writers
+{
+    public static class PrereleaseCompatibilityTransformer
+    {
+        // As the SARIF Technical Committee develops the SARIF specification, it
+        // releases incremental versions of the schema, with SemVer versions such as
+        // "2.0.0-csd.2.beta.2018-10-10". To avoid having to update the version strings
+        // in the test files every time this happens, we replace the version
+        // in the test file with the current version after reading the file
+        // into memory.
+        private const string VersionPropertyPattern = @"""version""\s*:\s*""[^""]+""";
+        private static readonly Regex s_VersionRegex = new Regex(VersionPropertyPattern, RegexOptions.Compiled);
+
+        private const string SchemaPropertyPattern = @"""\$schema""\s*:\s*""[^""]+""";
+        private static readonly Regex s_SchemaRegex = new Regex(SchemaPropertyPattern, RegexOptions.Compiled);
+
+        public static string UpdateToCurrentVersion(string prereleaseSarifLog, bool forceUpdate = false, Formatting formatting = Formatting.None)
+        {
+            bool modifiedLog = false;
+            JObject sarifLog = JObject.Parse(prereleaseSarifLog);
+
+            // Some tests update the semantic version to current for non-updated content. For this situation, we 
+            // allow the test code to force a transform, despite the fact that the provided version doesn't call for it.
+            string version = forceUpdate ? "2.0.0" : (string)sarifLog["version"];
+
+            switch (version)
+            {
+                // This case actually handles all SARIF v2 prerelease before we 
+                // started properly updating the schema. 
+                case "2.0.0":
+                {
+                    modifiedLog |= ApplyCoreTransformations(sarifLog);
+                    break;
+                }
+
+                case "2.0.0-csd.2.beta.2018-10-10":
+                {
+                    modifiedLog |= ApplyCoreTransformations(sarifLog);
+                    modifiedLog |= ApplyChangesSinceTechnicalCommitteeMeeting25();
+                    break;
+                }
+
+                default:
+                    break;
+            }
+
+            string text = modifiedLog ? sarifLog.ToString(formatting) : prereleaseSarifLog;
```

Comment:
>text [](start = 19, length = 4)

Or just dispense with the temporary and return the result of the ternary expression.